pkg/tsdb/prometheus/client/provider_test.go

```diff
@@ -0,0 +1,47 @@
+package client
```

Comment:
at one point you recommended using the `_test` suffix for tests. is there a reason why you didn't go with that here?

Reply:
When adding the `_test` suffix you only have access to the explicitly exported things.  These tests check unexpoted functions like `shouldForceGet`, so need to live in the same package name

use `_test` package when you want to test the external API

Reply:
@ryantxu yep, that part makes sense. i was looking for clarification on when the `_test` suffix is recommended. i thought @MasslessParticle's original recommendation was to always use it, and avoid testing unexported functions

Reply:
That's true, but the goal of this refactor was is to get things in packages where we can tackle them individually from there. Because the tests were written with the assumption that they have access to everything, that's it's own refactor.

Reply:
thanks for the clarification @MasslessParticle. makes sense 👍